.github/workflows/main.yml

```diff
@@ -45,14 +37,20 @@ jobs:
       - name: Archive artifacts
         uses: actions/upload-artifact@v2
         with:
-          name: AppImage i386
+          name: AppImage ${{ matrix.arch }}
           path: linuxdeploy-plugin-appimage*.AppImage*
+      - name: Upload release asset
```

Comment:
What's the point of introducing a second artifacts upload? Especially with some random third-party GitHub action (no offense, but I'm not a fan of that kind of stuff)?

Currently, the build artifacts are uploaded to GitHub's artifacts storage, then downloaded and a release is created using pyuploadtool. This seems redundant, and apparently introduces a potential race condition.

I guess this could be a misunderstanding: pyuploadtool handles the release flow already sufficiently well. One does not have to create releases and have the pipeline run then, for instance. Instead, creating a tag triggers the workflow, which then runs pyuploadtool, which creates a release using the GitHub API and uploads the specified artifacts.

Reply:
The reason that I use this workflow is that it allows me to add handwritten release notes in the web interface. Personal preference.

If you prefer to have all without any web interface interaction, the approach with pyuploadtool sounds just fine and I can remove this part of the PR.

Reply:
I'd prefer the pyuploadtool workflow indeed, so please do so. Thanks!

Reply:
done